gandalf.js

```diff
@@ -1,16 +1,17 @@
 /**
```

Comment:
SCRIPT-The way you've refactored this complex method into simpler ones is great. It makes the code much more maintainable.